testing/logging/test_fixture.py

```diff
@@ -54,6 +75,38 @@ def test2(caplog):
     result.stdout.no_fnmatch_line("*log from test2*")
 
 
+def test_change_disabled_level_undo(pytester: Pytester) -> None:
+    """Ensure that 'force_enable_logging' in 'set_level' is undone after the end of the test.
+
+    Tests the logging output themselves (affected by disabled logging level).
+    """
+    pytester.makepyfile(
+        """
+        import logging
+
+        def test1(caplog):
+            logging.disable(logging.CRITICAL)
+            caplog.set_level(logging.INFO)
+            # using + operator here so fnmatch_lines doesn't match the code in the traceback
+            logging.info('log from ' + 'test1')
+            assert 0
+
+        def test2(caplog):
+            # using + operator here so fnmatch_lines doesn't match the code in the traceback
+            # use logging.warning because we need a level that will show up if logging.disabled
+            # isn't reset to ``CRITICAL`` after test1.
+            logging.warning('log from ' + 'test2')
+            assert 0
+    """
+    )
+    result = pytester.runpytest()
+    result.stdout.fnmatch_lines(["*log from test1*", "*2 failed in *"])
+    result.stdout.no_fnmatch_line("*log from test2*")
+
+    # logging.disable needs to be reset because it's global and causes future tests will break.
+    logging.disable(logging.NOTSET)
```

Comment:
Since this needs to run, I’d make it a little fixture, so that logging gets restored even if the test exits early due to a failed assert.

Reply:
Oh that's a good idea. Good catch. I'll try to figure that out

Reply:
fixed